grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -766,6 +766,10 @@ class OrderController {
             if (!orderService.isOrderEditable(orderItem.order, session.user)) {
                 throw new UnsupportedOperationException("${warehouse.message(code: 'errors.noPermissions.label')}")
             }
+            if (params.quantity && orderItem.quantity != (params.quantity as Integer)) {
+                // if existing item's quantity is edited we have to trigger the order summary refresh
+                orderItem.disableRefresh = false
```

Comment:
can you explain why this is needed? Instinctively, I'd assume that if the orderItem's quantity is updated, the view would know that it'd need to be refreshed because a field is changing. Why does quantity need a special check here? If other params (ie not quantity) are updated, does the view get refreshed?

Reply:
> the view would know that it'd need to be refreshed because a field is changing.

That's correct for the view, but we have a materialized view for storing order summary (effectively a table in db).
Refreshing order summary for order **items** changes is disabled by default. It's refreshed only when a new item is added to the order. Item's quantity edit here did not trigger refresh on the order itself. In a result, the view has/calculates proper values, but the materialized view has old data, hence need to trigger this one at this level.

> Why does quantity need a special check here? If other params (ie not quantity) are updated, does the view get refreshed?

Only quantity matters for the order summary. Hence no need to trigger it after editing other properties